### PR TITLE
gtimelog: unstable-2020-05-16 -> unstable-2023-10-05 & cleanup

### PR DESCRIPTION
**Pull request by nixdrin**
## Description of changes

Upgrade gtimelog to latest master branch, move it to a Python application rather than library and clean up/modernize the definitions to be more similar to other GNOME-related Python applications.

I tried to address all review comments that were mentioned in #197545 (which does not seem to be actively updated anymore).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Cc: @jtojnar @SuperSandro2000

### CODE REVIEW

**Review by akgrant43 — APPROVED** [review]
I'm now running this PR on NixOS 23.05 x86_64-linux:

- functionality is working fine (based on ad-hoc testing).
- it's nice to have the .desktop file properly defined.
- the derivation appears to be cleaner.

Should the code be run through a formatter such as [alejandra](https://github.com/kamadorueda/alejandra)? (based on CONTRIBUTING.md stating that attribute sets that span multiple lines should be formatted one element per line).

Thanks!

**Review by oxzi — APPROVED** [review]
LGTM.

Please feel free to add yourself as another maintainer as I don't use `gtimelog` anymore for years.

**Comment by nixdrin** [comment]
@akgrant43 @oxzi Thanks for the review! I pushed an extra commit to fix the formatting of the `buildInputs` line. I think the rest of the file should be fine.

I made the changes in this PR because I wanted to try out gtimelog but it was somewhat broken in 23.05. If it ends up working well for my use case I'll consider adding myself as maintainer in the future. :)

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4756,8 +4756,6 @@ self: super: with self; {
 
   gtfs-realtime-bindings = callPackage ../development/python-modules/gtfs-realtime-bindings { };
 
-  gtimelog = callPackage ../development/python-modules/gtimelog { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4759]: This should get an entry in python-aliases.nix

File: pkgs/applications/office/gtimelog/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  buildInputs = [ glibcLocales gtk3 libsoup_3 libsecret ];
+  propagatedBuildInputs = with python3Packages; [
+    pygobject3
+  ];
+  checkInputs = with python3Packages; [
+    freezegun
+  ];
+
+  checkPhase = ''
+    substituteInPlace runtests --replace "/usr/bin/env python3" "${python3Packages.python.interpreter}"
```
Comment (SuperSandro2000) [inline pkgs/applications/office/gtimelog/default.nix:26]: We should use patchShebang here

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (nixdrin) [reply pkgs/top-level/python-packages.nix:4767]: Thanks! I added `gtimelog = throw "gtimelog has been promoted to a top-level attribute"; # added 2023-10-10`. I hope that's the correct entry, wasn't entirely sure. :)

File: pkgs/applications/office/gtimelog/default.nix (hunk 2 above)
Comment (nixdrin) [reply pkgs/applications/office/gtimelog/default.nix]: Fixed, thanks for the hint :)

**Review by panicgh — APPROVED** [review]
LGTM and program seems to work fine.

Result of `nixpkgs-review pr 259395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gtimelog</li>
    <li>gtimelog.dist</li>
  </ul>
</details>

**Comment by nixdrin** [comment]
Fixed the trivial conflict in `pkgs/top-level/python-aliases.nix`. No other changes.

**Review by panicgh — APPROVED** [review]
Application still works for me.

Result of `nixpkgs-review pr 259395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gtimelog</li>
    <li>gtimelog.dist</li>
  </ul>
</details>

**Comment by akgrant43** [comment]
I'm also still running the modified PR with no issues.

**Comment by nixdrin** [comment]
Fixed another trivial conflict, this time in `gtimelog/default.nix`. I kept the `format = "setuptools";` line that was added in 02dab4ab5ce6cc7a03736f3937650a161f5b3ae4. No other changes.

File: pkgs/top-level/python-aliases.nix [hunk 3]
```diff
@@ -178,6 +178,7 @@ mapAliases ({
   graphite_beacon = throw "graphite_beacon was removed, because it is no longer maintained"; # added 2022-07-09
   grappelli_safe = grappelli-safe; # added 2023-10-08
   grpc_google_iam_v1 = grpc-google-iam-v1; # added 2021-08-21
+  gtimelog = throw "gtimelog has been promoted to a top-level attribute"; # added 2023-10-10
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-aliases.nix:181]:
```suggestion
  inherit (pkgs) gtimelog;
```
this should work

Comment (nixdrin) [reply pkgs/top-level/python-aliases.nix]: Changed, thanks. I have to admit I'm not entirely sure how to test that it actually works. Hopefully some automatic check will complain if it does not :)

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by nixdrin** [comment]
I rebased and updated PR title and commit messages for more clarity, since the `unstable-2023-10-05` I'm updating to is not quite the latest commit on `master` anymore. But there also haven't really been significant changes since then in gtimelog that would warrant an update. The code changes itself are still identical. It's still working great and much better than before this PR. It should be good to go :)

I'm not actively using gtimelog anymore at the moment, so I'll already note that I might forget to keep checking for conflicts in this PR at some point. Sorry about that.

**Review by panicgh — APPROVED** [review]
Result of `nixpkgs-review pr 259395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gtimelog</li>
    <li>gtimelog.dist</li>
  </ul>
</details>

**Comment by infinisil** [comment]
You can ignore the `plgs/by-name` check failure, it's a bug to be fixed, merging won't cause problems :) 
(Though you can also move the package to `plgs/by-name` if you want)

**Comment by nixdrin** [comment]
Fixed another trivial conflict in `pkgs/top-level/python-packages.nix`. The "pkgs/by-name" GitHub check then started failing for some reason beyond my understanding. I managed to make it happy by moving gtimelog to `pkgs/by-name`. No other changes, still works exactly the same as before.

**Comment by nixdrin** [comment]
> You can ignore the `plgs/by-name` check failure, it's a bug to be fixed, merging won't cause problems :) (Though you can also move the package to `plgs/by-name` if you want)

Thanks for the clarification! I'll keep it in the moved place since I already tried that as a fix :D

**Comment by infinisil** [comment]
Btw this kind of `pkgs/by-name` check failure is fixed for future PRs now that https://github.com/NixOS/nixpkgs/pull/284765 is merged :)

**Pull request merged**